Update global_cycle program to apply soil temperature increments from GSI.

ClaraDraper-NOAA opened the pull request:
New feature to add capacity to read in soil temperature increments from an external file, generated by the GSI, then apply the increments to the model states. Closely follows the NST approach. Changes should be zero diff with current system, and have passed the regression test on hera. 

Note: There's a namelist change to retrieve the legacy setup, with adjt_nst_only replaced by do_sfccycle. 

Fixes #300.

ClaraDraper-NOAA committed:
Added capacity to read-in, and add soil increments to model prognostics.

ClaraDraper-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into global_2mDA

ClaraDraper-NOAA committed:
Updated scripts with new namelist options for calling sfccycle and lndinc.
This commit references issue #300.

ClaraDraper-NOAA committed:
Updated comments.

ClaraDraper-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/global_2mDA

ClaraDraper-NOAA committed:
Updated comments for Doxygen.

ClaraDraper-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/global_2mDA

ClaraDraper-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/global_2mDA

ClaraDraper-NOAA committed:
Merge branch 'feature/global_2mDA' of https://github.com/ClaraDraper-NOAA/UFS_UTILS into feature/global_2mDA

ClaraDraper-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/global_2mDA

GeorgeGayno-NOAA commented:
@barlage Who on the land team can serve as another reviewer?

barlage commented:
@GeorgeGayno-NOAA, @jiaruidong2017 is a good choice to review.

GeorgeGayno-NOAA commented:
@jiaruidong2017 I added you as a reviewer.

edwardhartnett commented on sorc/global_cycle.fd/cycle.f90:
```diff
@@ -7,40 +7,33 @@
+!!  2) Update the land states with increments read in from file (do_lndinc = .true.) 
+!!  3) Update the NSST field, several options: 
 !!
-!!  2) Update the surface fields and NSST TREF field using
```
Can you use "2." instead of "2)"? And the same for the other numbers. Then doxygen will understand you mean a numbered list.

ClaraDraper-NOAA replied:
Sure. I'll update that now.

ClaraDraper-NOAA replied:
Done.

ClaraDraper-NOAA replied:
Ed, can you please mark this change as done? I have a message saying merging is blocked until this change is made.

edwardhartnett commented:
How is it that you can add so much code and not change the output of the regression tests? Or did I misunderstand that?

Do you have a test which runs this code?

ClaraDraper-NOAA commented:
> How is it that you can add so much code and not change the output of the regression tests? Or did I misunderstand that?
> 
> Do you have a test which runs this code?

ClaraDraper-NOAA commented:
> How is it that you can add so much code and not change the output of the regression tests? Or did I misunderstand that?
> 
> Do you have a test which runs this code?

None of the new code is exercised if DO_LNDINC is .false. 

There's a test case here (on hera): 
/scratch2/BMC/gsienkf/Clara.Draper/DA_test_cases/2mDA/add_incr

(give me two minutes to fix the fix file links).

ClaraDraper-NOAA committed:
Tidy up numbering in comments for doxygen.

ClaraDraper-NOAA committed:
Reverting driver.hera.sh to original version.

GeorgeGayno-NOAA commented on sorc/global_cycle.fd/cycle.f90:
```diff
@@ -502,6 +525,51 @@ SUBROUTINE SFCDRV(LUGB, IDIM,JDIM,LENSFC,LSOIL,DELTSFC,  &
+
+    INQUIRE(FILE=trim(LND_FILE), EXIST=file_exists)
+    IF (.not. file_exists) then 
+        print *, 'ERROR: land increment update requested, but file does not exist: ', & 
```
@ClaraDraper-NOAA NCEP Central Operations (NCO) requires all messages associated with job termination to begin with "FATAL ERROR".  I got in trouble for this recently (see #368).

ClaraDraper-NOAA replied:
Just fixed that, and pushed it.

GeorgeGayno-NOAA commented on sorc/global_cycle.fd/read_write_data.f90:
```diff
@@ -1063,20 +1072,25 @@ SUBROUTINE NETCDF_ERR( ERR, STRING )
  !! native grid assumes no pole points, so these are removed.
  !!
  !! @param[in] GSI_FILE Path/name of the GSI file to be read.
+ !! @param[in] FILE_TYPE file-type to be read in, 'NST' or 'LND'.
```
I have not worked with the GSI lately.  The NST and land increments are in separate files?  And they are on the same gaussian grid?  Because I see variables IDIM_GAUS and JDIM_GAUS are used for both.

ClaraDraper-NOAA replied:
I don't know anything about the NST increments I'm afraid. The land increments are created by the GSI on the Gaussian grid, using a branch that Jeff Whitaker currently owns (he hasn't merged it back to develop yet, and it probably has a few minor edits before its ready). I checked that the grid info in the land increment file is the same as that generated within the code for the NST files, and it was.

GeorgeGayno-NOAA commented on sorc/global_cycle.fd/cycle.f90:
```diff
@@ -970,6 +1038,285 @@ SUBROUTINE ADJUST_NSST(RLA,RLO,SLMSK_TILE,SLMSK_FG_TILE,SKINT_TILE,&
+
+! CALCUALATE WEIGHTED INCREMENT OVER NEARBY GRID CELLS THAT HAVE SOIL
+
+! DRAPER: TO-DO, CODE ADDING INCREMENTS TO SOIL MOISTURE. 
```
After the increment is applied, how much will the soil temperature change?  Is the adjustment larger in the upper or lower layers?

ClaraDraper-NOAA replied:
The increments are generated by an EnKF, so the exact pattern depends on the EnKF inputs, but in general they decrease rapidly with depth, and are fairly small in an individual time step (< 0.5 K at the most in the surface layer). The soil EnKF is still being refined, and one of the issues I need to look at is whether to switch off the increments in the deepest layers.

GeorgeGayno-NOAA replied:
Ok.  Any change in soil temperature will change the percentage of frozen soil moisture.  A change <0.5K for cold layers may be small enough to not require an adjustment.  If a layer rises above freezing, the liquid soil moisture could be set to the total soil moisture.  @jiaruidong2017 do you have an opinion?

ClaraDraper-NOAA replied:
Good catch, George. For now, I'm not creating increments near freezing (or when the atmosphere is near freezing), but I think we should address this anyway.

GeorgeGayno-NOAA replied:
@HelinWei-NOAA may want to comment as well.

ClaraDraper-NOAA replied:
An issue here is that we're going to have to pull in model-specific code to make these consistency corrections to other variables after the land DA update (the same issue is going to come up when we add the snow increments). Does anyone have a good idea for how to handle this? I could write Noah model-specific consistency routines, and then later Noah-MP specific consistency routines. Other land DA systems I've worked with call routines along the lines of 'if soil temperature updated, check if soil liquid/ice needs to be updated' How do we feel about this code being inside UFS_UTILS/global_cycle though?  Is global_cycle already Noah-specific? @barlage

jiaruidong2017 commented:
I am wondering how the soil temperature increment files were generated by the GSI EnKF. What is the basic  theory or method you have to generate these increments?

ClaraDraper-NOAA commented:
> I am wondering how the soil temperature increment files were generated by the GSI EnKF. What is the basic theory or method you have to generate these increments?

I'm not quite sure what you're asking. The increments are created using the EnKF, following the standard EnKF theory. I added code to the gsi to properly generate H(x) for T2m, and then to add the soil temperature to the control vector. I've also added code to the stochastic physics / ufs community model to generate reasonable ensemble covariances across the land/atmos interface.

jiaruidong2017 commented:
> > I am wondering how the soil temperature increment files were generated by the GSI EnKF. What is the basic theory or method you have to generate these increments?
> 
> I'm not quite sure what you're asking. The increments are created using the EnKF, following the standard EnKF theory. I added code to the gsi to properly generate H(x) for T2m, and then to add the soil temperature to the control vector. I've also added code to the stochastic physics / ufs community model to generate reasonable ensemble covariances across the land/atmos interface.

The reason I am asking this question is that the created increments are key in updating the soil temperature, and we want to understand. For example, the soil temperature responses to the T2m are different from the surface layer and deep layers, and the soil temperature changes at deep layers show the resistance to the T2m variations.  I believe you have considered these in your methods.

GeorgeGayno-NOAA commented:
@ClaraDraper-NOAA I was able to run your test case using 41a562b. But I had to define LND_FILE in some scripts:

```
--- a/ush/global_cycle.sh
+++ b/ush/global_cycle.sh
@@ -96,6 +96,8 @@
 #                   Defaults to ${FIXam}/seaice_newland.grb
 #     NST_FILE      GSI file on the gaussian grid containing NST increments.
 #                   Defaults to NULL (no file).
+#     LND_FILE      GSI file on the gaussian grid containing land temperature increments.
+#                   Defaults to NULL (no file).
 #     FNTSFA        Input SST analysis GRIB file.
 #                   Defaults to ${COMIN}/${PREINP}sstgrb${SUFINP}
 #     FNACNA        Input sea ice analysis GRIB file.
@@ -291,6 +293,7 @@ FNVMXC=${FNVMXC:-${FIXam}/global_shdmax.0.144x0.144.grb}
 FNSLPC=${FNSLPC:-${FIXam}/global_slope.1x1.grb}
 FNMSKH=${FNMSKH:-${FIXam}/global_slmask.t1534.3072.1536.grb}
 NST_FILE=${NST_FILE:-"NULL"}
+LND_FILE=${LND_FILE:-"NULL"}
 FNTSFA=${FNTSFA:-${COMIN}/${PREINP}sstgrb${SUFINP}}
 FNACNA=${FNACNA:-${COMIN}/${PREINP}engicegrb${SUFINP}}
 FNSNOA=${FNSNOA:-${COMIN}/${PREINP}snogrb${SUFINP}}
@@ -388,6 +391,7 @@ EOF
 cat << EOF > fort.37
  &NAMSFCD
   NST_FILE="$NST_FILE",
+  LND_FILE="$LND_FILE"
  /
 EOF
```

and here:

```
--- a/ush/global_cycle_driver.sh
+++ b/ush/global_cycle_driver.sh
@@ -52,6 +52,7 @@ fi

 export DO_SFCCYLE=${DO_SFCCYCLE:-".true."}
 export DO_LNDINC=${DO_LNDINC:-".false."}
+export LND_FILE=${LND_FILE:-"NULL"}

 CRES=$(echo $CASE | cut -c 2-)
 JCAP_CASE=$((2*CRES-2))
```

ClaraDraper-NOAA commented:
> > > I am wondering how the soil temperature increment files were generated by the GSI EnKF. What is the basic theory or method you have to generate these increments?
> > 
> > 
> > I'm not quite sure what you're asking. The increments are created using the EnKF, following the standard EnKF theory. I added code to the gsi to properly generate H(x) for T2m, and then to add the soil temperature to the control vector. I've also added code to the stochastic physics / ufs community model to generate reasonable ensemble covariances across the land/atmos interface.
> 
> The reason I am asking this question is that the created increments are key in updating the soil temperature, and we want to understand. For example, the soil temperature responses to the T2m are different from the surface layer and deep layers, and the soil temperature changes at deep layers show the resistance to the T2m variations. I believe you have considered these in your methods.

Yes. In the EnKF the model error covariances from the ensemble determine the increments to each layer - so the increments in each layer will reflect the physical relationship between errors in that layer and errors in the T2m. I don't need to explicitly account for this in my method, the EnKF handles it. Does this answer your question?

edwardhartnett commented:
@ClaraDraper-NOAA there are lots of questions asked and answered above. Can you also put that information into the doxygen comments in the code, so that future users of this code can know those answers?

ClaraDraper-NOAA commented:
> @ClaraDraper-NOAA there are lots of questions asked and answered above. Can you also put that information into the doxygen comments in the code, so that future users of this code can know those answers?

I expanded the comments: 
!!  2. Update the land states with increments read in from file (do_lndinc = .true.)
!!     Designed to work with a land increment file  created by the GSI on the Gaussian
!!     grid. The increments are interpolated here to the model grid, using the
!!     same method as for the NST increments. Initially implemented for
!!     applying soil temperature increments calculated from the EnKF
!!     assimilation of T2m (but this is not a requirement -  any
!!     GSI-generated soil temperature increment file can be applied here).

I don't think an in depth discussion of the EnKF is needed here. In fact, I don't like the last sentence of my comment. This code block is intended to read in and apply a soil temperature increment file, and is (intentionally) not specific to the DA methodology used to generate that file, nor the obs assimilated. The DA methodology will very likely change over time.

ClaraDraper-NOAA committed:
Additional comments.

ClaraDraper-NOAA committed:
Bug fix to driver scripts.

ClaraDraper-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/global_2mDA

ClaraDraper-NOAA committed:
Pulled land increment code into separate module.
Changed input file name for soil increments, to distinguish from (later) snow increment file.
Added framework for updating smc after stc DA update.

ClaraDraper-NOAA committed:
Added noah routines for reading in soil params.

ClaraDraper-NOAA committed:
Created separate directory for LSM routines needed in land DA updates
and moved noah routines into noah subdirectory, to be compiled as a
library. Added frh2o routine from noah to recalculate liquid/frozen soil
moisture after making soil temperature DA update.

ClaraDraper-NOAA committed:
Created separate directory for LSM routines needed in land DA updates
and moved noah routines into noah subdirectory, to be compiled as a
library. Added frh2o routine from noah to recalculate liquid/frozen soil
moisture after making soil temperature DA update.

ClaraDraper-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/global_2mDA

ClaraDraper-NOAA committed:
Minor clean-up, added compile-time constant to limit number of soil
layers updated.

ClaraDraper-NOAA committed:
Merge branch 'feature/global_2mDA' of https://github.com/ClaraDraper-NOAA/UFS_UTILS into feature/global_2mDA

ClaraDraper-NOAA committed:
Added doxygen for lsm_routines and noah directories.

ClaraDraper-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/global_2mDA

ClaraDraper-NOAA commented:
After consulting with Mike Barlage and George Gaynor, I've added an additional directory to contain the land surface model code necessary to make secondary adjustments after the land DA updates, and have added a folder for the NOAH LSM with the necessary routines for updating soil moisture after updating soil temperature. Where possible, I've copied files from the NOAH model unchanged, however the NOAH model is not setup to make the necessary routines available externally, so I've had to create one new file with routines edited and copied from NOAH (Noah will soon be replaced with Noah-MP, and we thought this is good enough as a temporary solution).

ClaraDraper-NOAA committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/UFS_UTILS into feature/global_2mDA

ClaraDraper-NOAA committed:
removed -DENABLE_DOCS=ON in cmake flags

ClaraDraper-NOAA committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/UFS_UTILS into feature/global_2mDA

ClaraDraper-NOAA committed:
Clean-up after merge.

ClaraDraper-NOAA committed:
Added comments missed in merge to global_cycle

ClaraDraper-NOAA committed:
Fixed doxygen warnings in new code

ClaraDraper-NOAA committed:
Fixed doxygen warnings in new code

ClaraDraper-NOAA committed:
Merge branch 'feature/global_2mDA' of https://github.com/ClaraDraper-NOAA/UFS_UTILS into feature/global_2mDA

ClaraDraper-NOAA committed:
replace .eq. with .eqv.

ClaraDraper-NOAA committed:
removed Noah files with multiple doxygen problems.

ClaraDraper-NOAA committed:
More doxygen issues.

ClaraDraper-NOAA committed:
minor changes to get .f files passed github checks

ClaraDraper-NOAA committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/UFS_UTILS into feature/global_2mDA

ClaraDraper-NOAA committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/UFS_UTILS into feature/global_2mDA

ClaraDraper-NOAA committed:
edits for Doxygen compilation of f77 files.

ClaraDraper-NOAA committed:
changes to sflx_snippet for doxygen

ClaraDraper-NOAA committed:
Converted noah LSM f77 files to f90 for Doxygen.

ClaraDraper-NOAA committed:
fixing incomplete push

ClaraDraper-NOAA committed:
Converted sflx_snippet from f77 to f90

GeorgeGayno-NOAA commented:
@ClaraDraper-NOAA Please merge the latest updates from 'develop'.

ClaraDraper-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into feature/global_2mDA

ClaraDraper-NOAA commented:
> @ClaraDraper-NOAA Please merge the latest updates from 'develop'.

Done. Note, I'm in a workshop tomorrow and Wednesday, so will be slow responding.

GeorgeGayno-NOAA commented:
@ClaraDraper-NOAA Your test case in `/scratch2/BMC/gsienkf/Clara.Draper/DA_test_cases/2mDA` should be made into a consistency (regression) test. I can help with that.

ClaraDraper-NOAA commented:
> @ClaraDraper-NOAA Your test case in `/scratch2/BMC/gsienkf/Clara.Draper/DA_test_cases/2mDA` should be made into a consistency (regression) test. I can help with that.
Sure. What's the best way to design it? I see there's a reg_tests/global_cycle already. Should I add a new .sh with the new test into that folder? 

(if it's just for checking the code, the resolution of the test case doesn't really matter, so I can use the same restarts as used in the current global_cycle test if that is easier).

GeorgeGayno-NOAA reviewed (commented):
> > @ClaraDraper-NOAA Your test case in `/scratch2/BMC/gsienkf/Clara.Draper/DA_test_cases/2mDA` should be made into a consistency (regression) test. I can help with that.
> > Sure. What's the best way to design it? I see there's a reg_tests/global_cycle already. Should I add a new .sh with the new test into that folder?
> 
> (if it's just for checking the code, the resolution of the test case doesn't really matter, so I can use the same restarts as used in the current global_cycle test if that is easier).



> > @ClaraDraper-NOAA Your test case in `/scratch2/BMC/gsienkf/Clara.Draper/DA_test_cases/2mDA` should be made into a consistency (regression) test. I can help with that.
> > Sure. What's the best way to design it? I see there's a reg_tests/global_cycle already. Should I add a new .sh with the new test into that folder?
> 
> (if it's just for checking the code, the resolution of the test case doesn't really matter, so I can use the same restarts as used in the current global_cycle test if that is easier).

Yes, add a new .sh script under ./reg_tests. We just want to check the new option. If that can be done with the existing restarts, the use them. Set up the test to mimic its expected use in OPS (i.e., how will "DONST", "DO_SFCCYCLE" and "DO_LNDINC" be typically set?)

GeorgeGayno-NOAA commented:
@ClaraDraper-NOAA We are being encouraged to write unit tests for all new code. I wrote a simple test for your new land_increments module. I can point you to my changes. Or will you let me update your branch?

ClaraDraper-NOAA commented:
George,

Thanks for doing this.

I'm happy for you to update my branch. I'll then clone your test before
tomorrow morning, and have a look at it to make sure that it behaves/fails
as expected.

Clara



On Thu, Jun 17, 2021 at 2:42 PM GeorgeGayno-NOAA ***@***.***>
wrote:

> @ClaraDraper-NOAA <https://github.com/ClaraDraper-NOAA> We are being
> encouraged to write unit tests for all new code. I wrote a simple test for
> your new land_increments module. I can point you to my changes. Or will you
> let me update your branch?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/420#issuecomment-863550430>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH7BYDZF7GGRKOKYIJFMYG3TTJM3JANCNFSM4ZLIIZGQ>
> .
>


-- 
--------------------------------------------------------------------
Clara Draper
Physical Scientist
NOAA Earth System Research Laboratories
Physical Sciences Laboratory
325 Broadway, Boulder, CO
DSRC, Rm 1D-105
+1 303 497 4467
--------------------------------------------------------------------

GeorgeGayno-NOAA committed:
Fix syntax error in ./ush/global_cycle.sh

Fixes #300

GeorgeGayno-NOAA committed:
Place routine remap_coef in new utils module, which makes
writing unit tests easier.

Fixes #300.

GeorgeGayno-NOAA committed:
Add new unit test for land_increments.

Fixes #300

GeorgeGayno-NOAA committed:
Minor update.

Fixes #300

GeorgeGayno-NOAA commented:
@ClaraDraper-NOAA Everything is checked. Feel free to expand or change my test as you wish. We want a basic sanity check.

ClaraDraper-NOAA committed:
Added regression test for adding land increments.

ClaraDraper-NOAA commented:
I've added a reg_test (C768.lndinc.sh)  but I wasn't sure how to call two
tests from global_cycle so I haven't included the code to actually call it.

You'll need to copy the data
from /scratch2/BMC/gsienkf/Clara.Draper/ufs_utils_lndinc into your $HOMEreg

On Thu, Jun 17, 2021 at 3:31 PM GeorgeGayno-NOAA ***@***.***>
wrote:

> @ClaraDraper-NOAA <https://github.com/ClaraDraper-NOAA> Everything is
> checked. Feel free to expand or change my test as you wish. We want a basic
> sanity check.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/420#issuecomment-863577994>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH7BYD7MGYP53DZSSD56PI3TTJSTFANCNFSM4ZLIIZGQ>
> .
>


-- 
--------------------------------------------------------------------
Clara Draper
Physical Scientist
NOAA Earth System Research Laboratories
Physical Sciences Laboratory
325 Broadway, Boulder, CO
DSRC, Rm 1D-105
+1 303 497 4467
--------------------------------------------------------------------

GeorgeGayno-NOAA commented:
@ClaraDraper-NOAA Will take a look at your test today.

GeorgeGayno-NOAA commented:
@ClaraDraper-NOAA I updated the Hera driver script to run your test. May I check my changes into your branch?

ClaraDraper-NOAA commented:
sure. Thanks.

On Mon, Jun 21, 2021 at 1:43 PM GeorgeGayno-NOAA ***@***.***>
wrote:

> @ClaraDraper-NOAA <https://github.com/ClaraDraper-NOAA> I updated the
> Hera driver script to run your test. May I check my changes into your
> branch?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UFS_UTILS/pull/420#issuecomment-865296164>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH7BYDYXGKAXPPWNQYS2ATTTT6I55ANCNFSM4ZLIIZGQ>
> .
>


-- 
--------------------------------------------------------------------
Clara Draper
Physical Scientist
NOAA Earth System Research Laboratories
Physical Sciences Laboratory
325 Broadway, Boulder, CO
DSRC, Rm 1D-105
+1 303 497 4467
--------------------------------------------------------------------

GeorgeGayno-NOAA committed:
Update global cycle consistency test scripts for Hera.

Fixes #300

GeorgeGayno-NOAA committed:
Update Orion global_cycle consistency test script.

Fixes #300

GeorgeGayno-NOAA committed:
Update consistency test driver script on Jet.

Fixes #300

GeorgeGayno-NOAA committed:
Update .reg_tests/rt.sh for changes to the global_cycle
tests.

Fixes #300

GeorgeGayno-NOAA commented:
@ClaraDraper-NOAA I have the consistency tests working on Hera, Jet and Orion. But I can't test on WCOSS as it is undergoing maintenance for the next three days.

GeorgeGayno-NOAA committed:
Update global_cycle driver script on Dell.

Fixes #300

GeorgeGayno-NOAA committed:
Update global_cycle consistency test driver script for
Cray.

Fixes #300.

GeorgeGayno-NOAA committed:
Fix bug in Dell consistency test script.

Fixes #300

GeorgeGayno-NOAA commented:
@ClaraDraper-NOAA I would like to merge this next. Can you update your branch from 'develop'.

ClaraDraper-NOAA committed:
Merge branch 'develop' into feature/global_2mDA

GeorgeGayno-NOAA committed:
Merge branch 'develop' into feature/global_2mDA

GeorgeGayno-NOAA committed:
Add LSanSuppress.supp file for unit test.

Fixes #300.

ClaraDraper-NOAA commented:
> @ClaraDraper-NOAA I would like to merge this next. Can you update your branch from 'develop'.

I've updated it, pushed it, and the global_cycle consistency checks have passed.

GeorgeGayno-NOAA commented:
@ClaraDraper-NOAA Some tests are failing. I think I know what is going on. I will update your branch.

GeorgeGayno-NOAA commented:
> @ClaraDraper-NOAA Some tests are failing. I think I know what is going on. I will update your branch.

Some unit tests, I mean. The consistency checks are passing on all machines now.

GeorgeGayno-NOAA committed:
Update CMakeLists.txt to copy suppression file to
test directory.

Fixes #300.

GeorgeGayno-NOAA commented:
@ClaraDraper-NOAA I will merge this today.

GeorgeGayno-NOAA reviewed (commented):
Generated the doxygen for the new land routines - it looked good.

GeorgeGayno-NOAA reviewed (approved):
The Doxygen for the new routines was checked and looked good. The new unit test confirms the land update is working. The new consistency test is running on all supported machines. The existing consistency passed on all machines as expected (because the new land update option is turned off by default).

GeorgeGayno-NOAA commented:
@edwardhartnett It won't let me merge until your requested changes are addressed. Do you have any further comments?
